### DISCUSSION TITLE
Jekyll ou Medium?

### DISCUSSION THREAD

**Discussion by devigor** [frontendbr/forum#667] [Discussão]
#### Hi guys!!!

Vi que o Medium tem crescido bastante e gostaria de saber se vale a pena migrar um blog em Jekyll para a plataforma do Medium?
E quais são as vantagens ou desvantagens de migrar para essa plataforma

**Comment by aferreira44** [comment]
Também tenho a mesma dúvida. Acredito que em audiência o Medium ganhe, mas aí não perdemos a oportunidade dos leitores continuarem navegando e também de melhorar o SEO no nosso próprio site?

Seria uma má ideia publicar o mesmo conteúdo em ambas as plataformas?

**Comment by woliveiras** [comment]
Acredito que o único problema do Medium seja que o controle de tudo não está nas nossas mãos.

Imagina que, sobre a navegação dentro do site, se linkarmos outros de nossos posts (do Medium ou não) dentro de uma publicação no Medium nós fazemos com que os leitores continuem navegando em nosso conteúdo.

Porém, entretanto e todavia, o Medium é uma plataforma e nós não temos o controle de tudo. Meu principal medo é de que algum dia eles atualizem o contrato com os criadores de conteúdo e coloquem anúncio no meio do texto ou mesmo nas laterais, etc. Isso seria bom pra manter a empresa, porém ruim pra quem lê. - acho que ninguém que busca um conteúdo específico gosta de ficar vendo anúncio no meio do texto.

O Medium tem o esquema de newsletter que mandam para nossos seguidores, diminuindo um trampo nosso, também da pra seguir via RSS (não que não daria pra seguir usando static generators, é só um comentário aleatório).

O próprio Medium linka artigos legais pra pessoa continuar a leitura, mesmo que não sejam de nossa autoria, mas tem relação com o que a pessoa busca, o que eu acho legal.

Mas eu uso Jekyll, Medium e Hexojs e a única diferença entre eles - que eu sinto - é o editor de texto do Medium que é muito bom.

**Comment by lucianobarauna** [comment]
@devigor 

Eu tive essa dúvida também e escolhi o medium por conta dessas facilidades que o @willianjusten comentou. Agora em relação ao SEO é relativo ao meu ver porque de uma forma ou outra no final vc vai manter a presença na net do mesmo jeito.

Acho legal também você arranjar uma forma de linkar para o seu site o usuário do medium. Ex: Você pode saber mais sobre eu nesse link ( link para seu site)

**Comment by kvnol** [comment]
Eu amo o Medium e vou defendê-lo. 

O Medium é uma ferramenta fenomenal, com um alcance incrível por conta dos links em posts relacionados, mas concordo 100% com o @woliveiras e é um risco confiar numa plataforma que não está em nosso controle. 

Agora, uma solução que aprendi com o @lfeh é utilizar o Medium para posts mais metodológicos, ideológicos etc e o Jekyll, ou qualquer cms interno, para posts técnicos.

**Comment by wilcorrea** [comment]
Eu não consigo nem comparar os dois. O Medium é um plataforma repleta de limitações e cheia de comodidades. O Jekyll é uma aplicação cheia de liberdade, mas com pouca comodidade.

Logo, se você quer migrar tem que avaliar o que é que você espera do serviço em si.

Eu tenho preguiça infinita de manter infra. Mesmo Wordpress da vida me faria ter que instalar um tema (tenho que escolher antes ainda e só isso já levaria meses...); me achei com o Medium. Contudo pode ser que não seja a sua praia e você queira, por exemplo, add widgets ou ter controles de stats do seu blog que não tem na plataforma... nesse caso o melhor é você manter o assunto num ambiente que você domine.

(minha resposta é mais do mesmo, mas fica claro que é um assunto simples de se resolver @devigor ; )

**Comment by rubenmarcus** [comment]
simples, eu to montando um portfolio novo em que a sessão "blog" vou tirar da API do Medium, assim aproveito o alcance social do Medium, mas também por questões de SEO e de quem estiver já navegando no meu portfolio vai ver os mesmos artigos que estão no medium, só que no meu site.

https://developers.medium.com/

**Comment by lucianobragaweb** [comment]
Vou deixar aqui a experiência que estou tendo no momento com meu blog/site: [https://lucianobragaweb.github.io](https://lucianobragaweb.github.io) (Sim, eu sei que posso usar meu domínio personalizado).

Estou desenvolvendo meu Blog/Site com Jekyll. Acho fantástico, como também outros geradores de conteúdo estáticos. Uma coisa que notei na maioria dos comentários anteriores é sobre essa questão de quem está no controle. Isso é um assunto que rende, já vi palestras no W3C sobre isso e um cara falava inclusive sobre questões do tipo: Você ter e hospedar sua própria rede social pra evitar qualquer interferência externa e ter controle sobre seu conteúdo... coisas do tipo, que tem um certo sentido mas que na prática é pouco viável.

Então meu principal motivo em utilizar o Jekyll é justamente isso, estar no controle, inclusive no controle das versões. Meu Blog é hospedado no GitHub Pages, ou seja, é um repositório do Github, não pago hospedagem, logo se eu morrer ou ter amnésia em alguns meses uma hospedagem iria deixar o site offline, mas no Github pages o conteúdo vai ficar lá, não sei se eternamente, mas vai ficar. kkkk (**Só pra citar um exemplo mais bizarro possível**) ** Se acontecer um apocalipse zumbi eu já não sei.

Uma coisa que achei muito bom no Jekyll foi a facilidade com que consegui atingir os 100/100 pontos no **Google Page Speed**. [Mais Informações no Repositório](https://github.com/lucianobragaweb/lucianobragaweb.github.io) Você tem também total controle sobre SEO, que já foi falado e também sobre o layout e a maneira como você vai formatar seu site, criando páginas, Landing Pages, se for o caso, enfim... Controle!!

### Vantagens
* Markdown
Acho que markdown é a sintaxe ideal para escrever artigos. Sofri bastante nos tempos de Wordpress escrevendo e utilizando HTML dentro dos posts e isso com certeza deixa muito sujo e não indico a ninguém.
* Editor de textos
Talvez você até saiba, mas pra quem não sabe. Tem como escrever seus posts em um blog Jekyll hospedado no GitHub usando o próprio editor de textos do GitHub em Markdown e ele vai renderizar em seu blog. Então não ter um editor não é desculpa. Tem também plugins pra isso, já via para o Hexo, mas não gostei e evito usar plugins.

Se você está lendo agora em Agosto de 2017 provavelmente meu [site](lucianobragaweb.github.io) ainda está bem simples mas estou apenas começando.

Muito legal a ideia do @rubenmarcus, eu também já dei uma olhada na API do Medium e é bem simples. Uma outra coisa legal nesse sentido é trazer os Reviews da FanPage do facebook como depoimentos para o parte de Porfólio.

Tem muita coisa que pode ser dita ainda sobre o assunto, mas fico por aqui.

Espero ter contribuído com a discussão, até a próxima!!

Um cara que com certeza pode contribuir mais é o @willianjusten!!

**Comment by bernardodiasc** [comment]
Acho que se for só pra publicar artigos o Medium é mais recomendado. Pq é uma rede, vc posta nela e pode ter uma potencia muito maior na audiencia além de varias funcionalidades sociais uteis com tão pouco trampo.

Se for um site com conteúdos diversos, além de meros artigos, ai iria pra uma solução personalizada, mas não seria nenhum gerador de sites, pq já usei muito varios deles e não gostei, opiniao pessoal, desenvolvi meu proprio cms baseado em arquivos estaticos e markdown com frontmatter - vlw Jekyll - mas mesmo assim nao é gerador de site, é gerador de dados, o site eu faço na mao mesmo pq é isso que faço todo dia. 😄

**Comment by lucianobarauna** [comment]
@rubenmarcus
Penso em fazer o mesmo ;-)

**Comment by oliveriojnn** [comment]
@rubenmarcus @lucianobarauna 

A idéia é boa, mas tenham cuidado com as punições do Google em caso de SEO. 

Fazendo dessa forma o Google vai entender como conteúdo duplicado! E o fato dele ter sido publicado no Medium, teu site vai ser considerado plagio. Já vi algumas pessoas serem punidas dessa forma..

**Comment by rubenmarcus** [comment]
@oliveriojnn  esse é um bom questionamento. O medium é mais voltado pra galera da área. E um site pessoal seria pra "todos" num geral. Recruiters, curiosos, gente que não é da área. É algo a se pensar mesmo. Mas talvez por questões apenas de ter a facilidade você coloque no site pessoal com no follow sei lá. Se bem que chama visitas os artigos né. É complicado mesmo.

**Comment by bernardodiasc** [comment]
@rubenmarcus na verdade o Medium é uma plataforma de publicação de artigos de qualquer tema, se voce olhar bem, vai encontrar todos os assuntos possíveis la. Antes do Medium havia um serviço chamado http://www.posterous.com/ que tinha a mesma proposta. Cheguei a publicar alguns artigos no Posterous, abre lá e vê como está hoje em dia hehehe

**Comment by rubenmarcus** [comment]
@bernardodiasc sim eu sei, mas to falando de público. Em qualquer assunto que seja, eu vejo o público que lê e interage geralmente é da mesma área que o autor. Num site pessoal não.

**Comment by felipefialho** [comment]
O que tenho feito é colocar assuntos técnicos no meu site e assuntos gerais no Medium.

Tem funcionado bem.

**Comment by oliveriojnn** [comment]
@rubenmarcus Isso. Usar o rel="nofollow" nessa situação é uma boa saída. 

Acho a idéia do @LFeh boa. Artigos técnicos no próprio blog e não-técnicos no Medium. Um divulgando o outro.

**Comment by willianjusten** [comment]
Bom, galera já falou bastante acima, então não preciso me repetir muito, mas uma coisa extremamente importante que eu tenho a acrescentar é: `branding`.

No Medium você vê as pessoas falando "Viu o post lá do Medium?", isso fica super vago e acaba "diminuindo" o autor.

Quando você tem um site próprio (seja qualquer gerador), as pessoas falam sobre o "Post de Fulano", então é seu nome que é passado e com isso, as pessoas buscam pelo **seu** conteúdo, ao invés de artigos jogados pelo Medium.

**Comment by felipefialho** [comment]
@willianjusten Concordo, mas acho que dependendo do assunto, a escala do Medium é maior, já que naturalmente acaba atingindo pessoas fora do seu target.

**Comment by lucianobarauna** [comment]
haeiuaheiuah vcs agora me colocaram em dúvida
E essa abordagem é legal @LFeh estou começando a curtir, agora umas dúvidas:
- Caso você precise como você substitui a newsletter que o medium envia ?
- O medium limita escrever post técnicos demais e porquê vc optou separar ?

**Comment by felipefialho** [comment]
> Caso você precise como você substitui a newsletter que o medium envia ?

Nem sabia disso

> O medium limita escrever post técnicos demais e porquê vc optou separar ?

Não limita, optei porque prefiro atrelar meus posts técnicos ao meu site, como disse o @willianjusten

**Comment by willianjusten** [comment]
Pow, confesso que não acho que posts no Medium teriam um alcance maior que dentro do meu blog mesmo... Porque muito dos acessos acontecem devido ao share em redes sociais (que ocorre com/sem Medium) e posts antigos tem um acesso muito forte orgânico, principalmente por causa do tratamento adequado do SEO.

O que eu acho que pode mudar é:

- Você já possui um número de seguidores razoável? Continue com seu blog, é mais vantagem.
- Você não possui muitos seguidores? Vá para o Medium para captar seguidores e então, migre depois para perpetuar seu nome.

Mas aí podem falar: "Ahhh, mas não faço blog para ter nome.", e eu concordo, mas todo mundo gosta que seus artigos sejam lidos e devidamente creditados. O "fazer nome" é apenas consequência do bom trabalho, que acaba trazendo mais e mais leitores.

**Comment by bernardodiasc** [comment]
Acho que "fazer o nome" é **um dos** objetivos de se publicar artigos técnicos na internet. Mesmo se encarar como algo divertido, creio que isso tenha objetivo profissional pra maioria das pessoas, então vender-se é importante! 😄

**Comment by maclovin** [comment]
Medium é legal mas vocês já experimentaram o http://telegra.ph?
Obs: tem API.

**Comment by ghost** [comment]
Jekyll
